dapplet/src/index.ts

```diff
@@ -151,121 +145,214 @@ export default class TwitterFeature {
   };
 
   onProfileButtonClaimExec = async (profile, me) => {
-    const nearAccount = await this.identityService.getNearAccount('twitter/' + profile.id);
-    if (!nearAccount) return alert('You must link NEAR account before continue.');
-    // if (!isParticipant(nearAccount) && this._network === NearNetwork.MAINNET) {
-    //   return alert(
-    //     'As part of the closed testing, the withdrawal of tokens is available for ' +
-    //       'the first testers of the dapplet who have sent at least one transaction ' +
-    //       'on the testnet and feedback to Learn NEAR Club before November 25, 2021. ' +
-    //       'We will make it available to everyone soon. Stay tuned!',
-    //   );
-    // }
-
+    const { websiteName } = await this.getCurrentUserAsync();
     try {
+      const connectedAccounts = await Core.connectedAccounts.getConnectedAccounts(
+        profile.id,
+        websiteName.toLowerCase(),
+      );
+      const nearAccountsFromCA = connectedAccounts
+        .flat()
+        .filter((ca) => ca.id.indexOf('near/testnet') !== -1)
+        .map((a) => a.id.split('/')[0]);
+      if (nearAccountsFromCA.length === 0) {
+        alert(
+          'You must link your NEAR account to your Twitter account using the Connected Accounts service. Click |⋈ Link| button to continue.',
+        );
+        return;
+      }
+      const accountGId = profile.id + '/' + websiteName.toLowerCase();
```

Comment:
Can this logic be moved to a separate service to make the dapplet code cleaner?

Reply:
fixed